deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/SparkDl4jNetworkWrapper.scala

```diff
@@ -79,15 +115,28 @@ abstract class SparkDl4jModelWrapper[T, E <: SparkDl4jModelWrapper[T, E]](overri
         } else throw new RuntimeException("Vector size must be greater than 0")
     }
 
-    protected def output(vector: Vector) : Vector = network.predict(vector)
+    protected def output(vector: Vector) : Vector = {
+        val predicted = outputTensor(vector)
+        if (!predicted.isVector) {
+            log.warn("The output of the tensor is not a vector. Flattening the tensor to match Spark's api.")
+        }
+        Vectors.dense(flattenTensor(predicted))
+    }
+
+    def setTrainingStats(sparkTrainingStats: SparkTrainingStats) : this.type = {
+        this.trainingStats = sparkTrainingStats
+        this
+    }
+
+    def getTrainingStats : SparkTrainingStats = trainingStats
 
     protected def outputTensor(vector: Vector) : INDArray = getMultiLayerNetwork.output(MLLibUtil.toVector(vector))
 
     protected def outputFlattenedTensor(vector: Vector) : Vector = Vectors.dense(flattenTensor(outputTensor(vector)))
 
     protected[SparkDl4jModelWrapper] class SparkDl4jModelWriter(instance: SparkDl4jModelWrapper[T,E]) extends MLWriter {
         override protected def saveImpl(path: String): Unit = {
-            ModelSerializer.writeModel(network.getNetwork, path, true)
+            ModelSerializer.writeModel(network, path, true)
         }
```

Comment:
Hmm - this only handles local. I'm wondering if maybe we could have this be a bit more agnostic and write to hdfs where hdfs *could* be local?